src/client/Client.cc

```diff
@@ -3447,6 +3446,7 @@ void Client::check_caps(Inode *in, unsigned flags)
 
     if (now < in->hold_caps_until) {
       ldout(cct, 10) << "delaying cap release" << dendl;
+      delayed_caps.push_back(&in->cap_item);
```

Comment:
remove above code and change this code to:

if (!(flags & CHECK_CAPS_NODELAY)) {
  ldout(cct, 10) << "delaying cap release" << dendl;
  in->hold_caps_until = now
  in->hold_caps_until += cct->_conf->client_caps_release_delay;
  delayed_caps.push_back(&in->cap_item);
}

I think it's more clear

Reply:
greate point ! 👍